articles/create-a-node-cli/index.md

```diff
@@ -0,0 +1,265 @@
+    const randomAuthor = chalk.green(res.data.author);
+    const log = `${randomQuote} - ${randomAuthor}`;
+    console.log(log);
+}).catch((err)=>{
+    const errorLog = chalk.red(err);
+    console.log(errorLog);
+})
+```
+
+Note: We are using chalk to color our logs and get a colorful output. We setup an empty string called `url` so that when the user selects quotes, we load in the quote API URL and the same goes for jokes.
+
+For now, the code won't work because we're not calling any of the functions.
+
+## Understanding option/command/argument parsing
+
+A good example of option parsing is when we passed _-y_ to our _npm init_ command to set everything to default.
+
+The commands used in CLI applications often come with other options or flags. They extend the functionality of the application like in the example above.
+
+A common application of options is when asking for user input which determines the output of the application.
+
+Node.js comes with an in-built app interface for the process module called `process.argv`. It takes in the arguments passed, invokes a method and returns an array with all the arguments passed into the main process.
+
+Using _process.arg_ might be tedious and not beginner friendly so we will use [yargs](https://www.npmjs.com/package/yargs) to help us process the options.
+
+Yargs offers a lot of options that make it very powerful and widely used. Some of them include `.command()`, `.help()`, `.example()`, `.usage()` and others. To read more on the different capabilities of the options see the yargs [docs](https://github.com/yargs/yargs).
+
+We will be making use of the command option to allow us to invoke a function when the set command is triggered. The syntax for this is:
+
+```javascript
+const argv = require("yargs");
+argv.command(
+  "name of command",
+  "description of the command",
+  functionA,
+  functionB
+).argv;
+```
+
+Note: _functionA_ will define the command arguments, then _functionB_ will implement the function.
+
+In our project we need two commands, let's set them up
+
+```javascript
+const argv = require("yargs");
+argv.command(
+  "joke",
+  "Fetching your joke",
+  (yargs) => {},
+  (argv) => {}
+).argv;
+
+argv.command(
+  "quote",
+  "Fetching a quote",
+  (yargs) => {},
+  (argv) => {}
+).argv;
+```
+
+Finally, let's pass in our functions to enable the jokes and quotes to be displayed.
+
+```javascript
+const argv = require("yargs");
+argv.command(
+  "joke",
+  "Fetching your joke",
+  (yargs) => {},
+  (argv) => {
+    getJoke();
+  }
+).argv;
+
+argv.command(
+  "quote",
+  "Fetching a quote",
+  (yargs) => {},
+  (argv) => {
+    getQuotes();
+  }
+).argv;
+```
+
+To get a joke or quote run:
+
+```terminal
+node index.js joke
+```
+
+```terminal
+node index.js quote
+```
+
+## Finishing Up
+
+Our full code now looks like this:
+
+```javascript
+#!/usr/bin/env node
+
+const axios = require("axios");
+const chalk = require("chalk");
+const argv = require("yargs");
+
+argv.command(
```

Comment:
It's confusing that the order you explained these two code blocks (`argv.command` and `getJoke()`) are reversed.

Reply:
unfortunately I do not understand this issue, @louisefindlay23 can you please elaborate?

Reply:
In your article, @katungi you explained the `function getJoke()` code first and `argv.command` second. Readers may be confused that in your final example they appear in a different order.

Reply:
I get it now; I have switched the order in the final code snippet